libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -312,6 +315,32 @@ void AP_AHRS::reset_gyro_drift(void)
 #endif
 }
 
+/*
+  update state structure after each update()
+ */
+void AP_AHRS::update_state(void)
+{
+    state.primary_IMU = _get_primary_IMU_index();
+    state.primary_gyro = _get_primary_gyro_index();
+    state.primary_accel = _get_primary_accel_index();
+    state.primary_core = _get_primary_core_index();
+    state.wind_estimate_ok = _wind_estimate(state.wind_estimate);
+    state.EAS2TAS = dcm.get_EAS2TAS();
+    state.airspeed_ok = _airspeed_estimate(state.airspeed);
+    state.airspeed_true_ok = _airspeed_estimate_true(state.airspeed_true);
+    state.airspeed_vec_ok = _airspeed_vector_true(state.airspeed_vec);
+    state.quat_ok = _get_quaternion(state.quat);
+    state.secondary_attitude_ok = _get_secondary_attitude(state.secondary_attitude);
+    state.secondary_quat_ok = _get_secondary_quaternion(state.secondary_quat);
+    state.location_ok = _get_location(state.location);
```

Comment:
This change significantly changes the output of the AHRS when DCM is the current backend.

On a 50Hz Plane travelling at 15m/s this could make a 30cm difference, depending on when you get around to calling the method in the loop (and assuming we're taking up the entire timeslice :-)

I'm actually good with this behaviour change.

Reply:
yep, I'm happy with that